test/framework/cluster_proxy.go

```diff
@@ -107,6 +107,8 @@ type ClusterLogCollector interface {
 	// TODO: describe output folder struct
 	CollectMachineLog(ctx context.Context, managementClusterClient client.Client, m *clusterv1.Machine, outputPath string) error
 	CollectMachinePoolLog(ctx context.Context, managementClusterClient client.Client, m *expv1.MachinePool, outputPath string) error
+	// CollectInfrastructureLogs collects log the infrastructure.
+	CollectInfrastructureLogs(ctx context.Context, clusterName string, outputPath string) error
```

Comment:
we should probably pass in the mgmtClusterClient + maybe the Cluster object itself?

Looks like we don't need it but it would be consistent with the other funcs and other providers probably need at least the mgmtClusterClient

Reply:
I agree with the mgmtClusterClient, but would vote against the cluster object for now, as we don't have it already available where `CollectInfrastructureLogs` (and also `CollectMachineLog`) gets called.

Reply:
(but also ok in doing the get at the place where its called)

Reply:
Sounds reasonable. Would cluster namespace + name be possible?

Reply:
nevermind, adjusted to pass the cluster object.